error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/MemberOrdering.java

```diff
@@ -0,0 +1,187 @@
+      return Description.NO_MATCH;
+    }
+
+    return buildDescription(tree)
+        .addFix(swapMembersWithComments(membersWithComments, sortedMembersWithComments, state))
+        .setMessage(
+            "Members, constructors and methods should follow standard ordering. "
+                + "The standard ordering is: static variables, non-static variables, "
+                + "constructors and methods.")
+        .build();
+  }
+
+  private static boolean isStatic(VariableTree memberTree) {
+    Set<Modifier> modifiers = memberTree.getModifiers().getFlags();
+    return modifiers.contains(Modifier.STATIC);
+  }
+
+  private static boolean isConstructor(MethodTree methodDecl) {
+    return ASTHelpers.getSymbol(methodDecl).isConstructor();
+  }
+
+  private static boolean shouldBeSorted(Tree tree) {
+    return tree instanceof VariableTree
+        || (tree instanceof MethodTree && !ASTHelpers.isGeneratedConstructor((MethodTree) tree));
+  }
+
+  private static SuggestedFix swapMembersWithComments(
+      ImmutableList<ClassMemberWithComments> memberWithComments,
+      ImmutableList<ClassMemberWithComments> sortedMembersWithComments,
+      VisitorState state) {
+    SuggestedFix.Builder fix = SuggestedFix.builder();
+    for (int i = 0; i < memberWithComments.size(); i++) {
+      Tree originalMember = memberWithComments.get(i).member();
+      ClassMemberWithComments correct = sortedMembersWithComments.get(i);
+      /* Technically this check is not necessary, but it avoids redundant replacements. */
+      if (!originalMember.equals(correct.member())) {
+        String replacement =
+            Stream.concat(
+                    correct.comments().stream().map(Tokens.Comment::getText),
+                    Stream.of(state.getSourceForNode(correct.member())))
+                .collect(joining("\n"));
+        fix.merge(
+            SuggestedFixes.replaceIncludingComments(
+                TreePath.getPath(state.getPath(), originalMember), replacement, state));
+      }
+    }
+    return fix.build();
+  }
+
+  /** Returns the class' members with their comments. */
+  private static ImmutableList<ClassMemberWithComments> getMembersWithComments(
+      ClassTree classTree, VisitorState state) {
+    return classTree.getMembers().stream()
+        .map(
+            member ->
+                new ClassMemberWithComments(member, getMemberComments(state, classTree, member)))
+        .collect(toImmutableList());
+  }
+
+  private static ImmutableList<Tokens.Comment> getMemberComments(
+      VisitorState state, ClassTree classTree, Tree member) {
+    if (member.getKind() == Tree.Kind.METHOD
+        && ASTHelpers.isGeneratedConstructor((MethodTree) member)) {
+      return ImmutableList.of();
+    }
+
+    checkState(
+        state.getEndPosition(member) != -1,
+        "Member's end position is not available (-1).\n - member=[%s]\n - source=[%s]",
+        member,
+        state.getSourceForNode(member));
```

Comment:
100% overkill, but it would be nice to not forget about this case or verify that it can never happen.

Reply:
I ran into this exact issue while testing, to my knowledge this case can only happen for generated members (e.g. generated constructors)

Reply:
I was thinking that Lombok and tools like it could also behave like this.

Reply:
This makes two of us, I've applied changes assuming that `state.getEndPosition(member)` is only unavailable for members that are not in the source file.